### PR TITLE
add CountWhileInRange in MatchContent.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/MatchContent.cs [hunk 1]
```diff
@@ -107,6 +106,39 @@ public int CountUntilAny(char[] ch, int maxCount)
...
+            {
+                return 0;
+            }
+            var length = Math.Min(maxCount, Length);
+            if (Direction == MatchDirection.Forward)
+            {
+                for (int i = 0; i < length; i++)
+                {
+                    var ch = Text[StartIndex + i];
+                    if (start >= ch && ch <= end)
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/MatchContent.cs]: >>= [](start = 30, length = 2)

<=

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/MatchContent.cs [hunk 2]
```diff
@@ -107,6 +106,39 @@ public int CountUntilAny(char[] ch, int maxCount)
...
+                    }
+                }
+                return length;
+            }
+            else
+            {
+                for (int i = 0; i < length; i++)
+                {
+                    var ch = Text[StartIndex - 1 - i];
+                    if (start >= ch && ch <= end)
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/MatchContent.cs]: >>= [](start = 30, length = 2)

<=

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/MatchContent.cs [hunk 3]
```diff
@@ -149,7 +212,8 @@ public int CountWhileAny(char[] ch, int maxCount)
             {
                 for (int i = 0; i < length; i++)
                 {
-                    if (Array.IndexOf(ch, Text[StartIndex + i]) == -1)
+                    var ch = Text[StartIndex + i];
+                    if (start < ch && ch > end)
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/MatchContent.cs]: >< [](start = 30, length = 1)

<

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/MatchContent.cs [hunk 4]
```diff
@@ -160,7 +224,8 @@ public int CountWhileAny(char[] ch, int maxCount)
             {
                 for (int i = 0; i < length; i++)
                 {
-                    if (Array.IndexOf(ch, Text[StartIndex - 1 - i]) == -1)
+                    var ch = Text[StartIndex - 1 - i];
+                    if (start < ch && ch > end)
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/MatchContent.cs]: >< [](start = 30, length = 1)

>

**Pull request merged**